src/components/network-chart/NetworkChartRow.js

```diff
@@ -29,117 +30,109 @@ import type { NetworkPayload } from '../../types/markers';
   // a specific payload.
   +networkPayload: NetworkPayload,
 };
 
-// This component splits a network marker duration in 4 different phases,
+// This component splits a network marker duration in different phases,
 // and renders each phase as a differently colored bar.
-// 1. request queue
-// 2. request
-// 3. response
-// 4. response queue (This is not calculated. We assume it is the rest of the duration.)
 const NetworkChartRowBar = (props: NetworkChartRowBarProps) => {
-  const { marker, networkPayload } = props;
+  const { marker, networkPayload, markerWidth } = props;
   const { start, dur } = marker;
 
-  // A marker does not always contain the same set of networkPayload on the start of
-  // the connection.
-  const queueStart =
-    networkPayload.secureConnectionStart ||
-    networkPayload.tcpConnectEnd ||
-    networkPayload.connectStart ||
-    networkPayload.domainLookupStart ||
-    0;
-
-  // The default for width of the phases is always zero.
-  let requestQueue: number = 0;
-  let request: number = 0;
-  let response: number = 0;
-
-  // Timestamps from network markers aren't adjusted when processing
-  // the gecko profile format. This is the reason why the start timestamp of a
-  // marker can be after the start timestamp of the request.
-  // To prevent false networkPayload visualization, we need this workaround.
-  // See https://github.com/firefox-devtools/profiler/issues/1493 for more detail.
-  if (
-    networkPayload.requestStart &&
-    start < networkPayload.requestStart &&
-    networkPayload.responseStart &&
-    start < networkPayload.responseStart
-  ) {
-    request =
-      (networkPayload.responseStart - networkPayload.requestStart) / dur * 100;
-  }
-
-  if (
-    networkPayload.responseEnd &&
-    start < networkPayload.responseEnd &&
-    networkPayload.responseStart &&
-    start < networkPayload.responseStart
-  ) {
-    response =
-      (networkPayload.responseEnd - networkPayload.responseStart) / dur * 100;
-  }
-
-  if (queueStart && start < queueStart && queueStart > 0) {
-    requestQueue = (queueStart - start) / dur * 100;
-  }
+  const barParts = [];
+  let previousValue = start;
+  let previousName = 'startTime';
 
-  // When we keep the default values (=zero), the
-  // response takes the full width of the bar to be more visible.
-  if (requestQueue + request + response === 0) {
-    response = 100;
-  }
+  PROPERTIES_IN_ORDER.filter(
+    property => typeof networkPayload[property] === 'number'
+  ).forEach((property, i, { length }) => {
+    // We force-coerce the value into a number just to appease Flow. Indeed the
+    // previous filter ensures that all values are numbers but Flow can't know
+    // that.
+    const value = +networkPayload[property];
+    barParts.push({
+      left: (previousValue - start) / dur * markerWidth,
+      width: Math.max((value - previousValue) / dur * markerWidth, 1),
+      opacity: i / (length - 1),
```

Comment:
This is simple but this works fine for me. The small drawback is that values for the same "phases" will be different when the properties count is different. I think it's acceptable because the actual color is not important, what's important is that we see them properly.

Reply:
Non-blocking: I feel strongly that we should pre-compute stable colors for each phase.

Reply:
Observation: Ideally the colors would be labeled somehow in the tooltip, but that is beyond the scope of this PR.

Reply:
Blocking: Also this will break if we have a length of `1`, as the opacity would be a divide by zero error.

Reply:
ah yeah this is pretty bad. I think I'll compute the opacities once and for all, and just use the computed values here. This should solve all problems mentioned here (including the problem I mentioned myself).